### PR TITLE
Add changes from club session + improvements

### PR DESCRIPTION
**Pull request by MartinMarwad**
Add changes worked during club session, plus a few other improvements and cleanups.

### CODE REVIEW

File: src/app/(auth)/signup/page.js
Comment (Gurpranked) [inline src/app/(auth)/signup/page.js:1]: Small typo in the CSS Baseline import statement. Should be `import CssBaseline from '@mui/material/CssBaseline` as stated in the following MUI docs page: https://mui.com/material-ui/react-css-baseline/. 
Hope it helps

**Review by Gurpranked — CHANGES_REQUESTED** [review]
Please review the comment placed on the signup/page.js file to fix the current build errors. 
Overall valuable edits, just a small typo.

File: src/app/(auth)/signup/page.js
Comment (MartinMarwad) [reply src/app/(auth)/signup/page.js:1]: Latest commit 4a44ebe fixed this 👍

Comment (Gurpranked) [reply src/app/(auth)/signup/page.js:1]: Nice!

Comment (Gurpranked) [reply src/app/(auth)/signup/page.js:1]: I still see some errors with the deployed application, not sure if it's just on my end.

**Review by Gurpranked — APPROVED** [review]

File: src/app/(auth)/signup/page.js
Comment (Gurpranked) [reply src/app/(auth)/signup/page.js:1]: Nevermind, it was just Amplify taking longer than expected to build all the dependencies. Looks good!

**Review by andrew-aiken — APPROVED** [review]
Probably is good

**Pull request merged**